compiler/rustc_attr_data_structures/src/stability.rs

```diff
@@ -132,9 +132,10 @@ pub enum StabilityLevel {
     Stable {
         /// Rust release which stabilized this feature.
         since: StableSince,
-        /// Is this item allowed to be referred to on stable, despite being contained in unstable
-        /// modules?
-        allowed_through_unstable_modules: bool,
+        /// This is `Some` if this item allowed to be referred to on stable, despite being contained
+        /// in unstable modules. `Some(Some(sym))` indicates that that usage is deprecated, with the
+        /// given deprecation message.
+        allowed_through_unstable_modules: Option<Option<Symbol>>,
```

Comment:
Either use different fields or extract to a separated struct instead of using `Option<Option<Symbol>>`. I would prefer us using an enum, somewhat like `UnstableReason`.

Reply:
Yeah that's fair, I was lazy...